src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.CoreCLR.cs

```diff
@@ -118,6 +119,12 @@ public override string ToString()
 
             return m_toString;
         }
+
+        public override bool Equals(object? obj) =>
+            obj is RuntimeConstructorInfo ci && m_handle == ci.m_handle && ReferenceEquals(m_declaringType, ci.m_declaringType);
```

Comment:
```suggestion
            ReferenceEquals(this, obj) ||
            (MetadataUpdater.IsSupported && CacheEquals(obj));
```
Is there a reason why we are not doing this here - so that it looks the same as PropertyInfo and EventInfo?

Reply:
Adding `ReferenceEquals(this, obj)` did not improve perf, second have been updating `RuntimeConstructorInfo` same as `RuntimeMethodInfo` as it is a method, but that might wrong